code/game/objects/items/devices/pinpointer.dm

```diff
@@ -123,9 +123,10 @@
 	var/turf/T = get_turf(src)
 	var/zlevels = GetConnectedZlevels(T.z)
 	var/cur_dist = world.maxx+world.maxy
-	for(var/obj/item/radio/beacon/R in world)
-		if(!R.functioning)
+	for(var/B in get_valid_teleporter_beacons())
```

Comment:
`for(var/obj/item/radio/beacon/B in get_valid_teleporter_beacons())` would implicitly typecheck each value with no additional overhead since you're manually typechecking anyway.